src/private/emscripten_api/emscripten_indoor_map_entity_information_api.js

```diff
@@ -144,6 +160,140 @@ function EmscriptenIndoorMapEntityInformationApi(emscriptenApiPointer, cwrap, ru
 
         return output;
     };
+
+    var _GetOutlines = function(indoorMapEntityModelId) {
+        var Outlines = [];
+
+        var indoorMapEntityNumOfOutlinesBuf = _emscriptenMemory.createInt32Buffer(1);
+
+        var success = _indoorEntityInformationApi_TryGetNumberOfPolygons(
+            _emscriptenApiPointer,
+            indoorMapEntityModelId, 
+            indoorMapEntityNumOfOutlinesBuf.ptr
+        );
+
+        var indoorMapEntityNumOfOutlines = _emscriptenMemory.consumeBufferToArray(indoorMapEntityNumOfOutlinesBuf)[0];
+
+        if (!success)
+        {
+            return null;
+        }
+
+        for(var outlineIndex = 0; outlineIndex < indoorMapEntityNumOfOutlines; outlineIndex++)
```

Comment:
I don't understand why we are needing to loop + make multiple calls here. Surely to obtain outline polygons for all entities in this request, we need to
1. call a C method to populate a buffer with information about buffer sizes
2. allocate buffers
3. call a C method to populate those buffers.